public/js/app/leaderboards.js

```diff
@@ -51,185 +39,130 @@ function leaderboardUpdate(data) {
-    
 
     if (playerIndex <= 0) {
-      //console.log('I broke');
-      playerIndex -= (playerIndex) * -1;
-    } else {
-      console.log(` We didn't break! `);
+      console.log('I broke');
     }
     //Switches background color for even and uneven
-    if (currentDate >= data[playerIndex].date) {
-      next100Players--;
-      addNextPlayer++;
-      previousMissingPlayer++;
-      //console.log(`${data[playerIndex].label} didn't play recently`); 
-      
-      
+    if (backgroundColor === 'greyLeaderboard') {
+      backgroundColor = 'darkGreyLeaderboard';
     } else {
-      if (backgroundColor === 'greyLeaderboard') {
-        backgroundColor = 'darkGreyLeaderboard';
-      } else {
-        backgroundColor = 'greyLeaderboard';
-      }
-      // Gets the player data and inserts it
-
-      let rating = data[playerIndex].value.rating;
-      let winRate = data[playerIndex].value.wonGames / data[playerIndex].value.totalgames * 100;
-      document.getElementById('onePlayer').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}' > ${data[playerIndex].label}</li>`);
-      document.getElementById('oneRank').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${-1 * (playerIndex - data.length + addNextPlayer)}</li>`);
-      document.getElementById('oneRating').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${rating.toFixed(0)}</li>`);
-      document.getElementById('oneGamesAmount').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${data[playerIndex].value.totalgames}</li>`);
-      document.getElementById('oneWon').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${winRate.toFixed(1)}% </li>`);
-
+      backgroundColor = 'greyLeaderboard';
     }
+    // Gets the player data and inserts it into the li element
+    let rating = playerList[playerIndex].value.rating;
+    let winRate = playerList[playerIndex].value.wonGames / playerList[playerIndex].value.totalgames * 100;
+    document.getElementById('onePlayer').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}' > ${playerList[playerIndex].label}</li>`);
+    document.getElementById('oneRank').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${-1 * (playerIndex - playerList.length)}</li>`);
+    document.getElementById('oneRating').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${rating.toFixed(0)}</li>`);
+    document.getElementById('oneGamesAmount').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${playerList[playerIndex].value.totalgames}</li>`);
+    document.getElementById('oneWon').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${winRate.toFixed(1)}% </li>`);
```

Comment:
`function createLi(playerNumber) {
  let createLi = document.createElement('li');
  createLi.classList.add('leaderboardDelete', `${backgroundColor}`);
  let textPlayerName = document.createTextNode(`${playerList[playerNumber].label}`);
  createLi.appendChild(textPlayerName);
  let leaderboardPlayer = document.getElementById(`leaderboardPlayer`);
  leaderboardPlayer.appendChild(createLi);
}`

Are you sure this is a better idea? I would need this about 10 times, adding an obscene amount of code to the js file. Seems like a very complicated way of dealing with such a simple issue. Not sure if there is a better way to do this since I haven't touched child and these sort of things ever

Reply:
why would you need this 10 times, does something prevent you from passing the changing parameters as part of the function arguments? :v

Reply:
But then I would need to call the function 10 times if I need to change the parameters.

Reply:
Which is okay, 10 function calls will take 10 lines at most, not really an "obscene amount of code" :p 
You can do it all in a function called "RefreshLeaderboardPages" or whatever, it's fine

Reply:
well yeah, I meant doing 10 times the components of the function but calling it many times would work aswell. Also do you know where could I find more about how the insertadjacent html lags a bunch? I couldnt find much but then again, I'm not very knowledgable about this stuff

Reply:
You can look for performance comparisons online and you'll see that it's slower because it requires parsing. From experience I know modifying / injecting HTML code directly in the page can refresh an undisclosed amount of DOM elements at once, which in the past has caused me extreme lag in some cases. It's just worse than modifying the DOM tree specifically with precises operations, so as it takes the same amount of space to factorize it & since it's also better practice I would enjoin you to do it. But it's your code on your branch, you do you 😁  As long as this never gets into develop it's fine

Reply:
Here's a page I found by googling a bit https://medium.com/codex/what-is-faster-to-insert-into-dom-html-or-dom-nodes-ff11586f8570 but there's a lot of benchmrking on webjs stuff online so feel free to look for other comparisons!